### PR TITLE
Migrate protein database actions to Protein module

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=49958

#### Changes
- Move all protein database actions, forms, and JSPs to Protein module: Protein admin, FASTA load/reload/delete, protein annotations, GO load, ProtSprotOrgMap
- Split out `SimpleProtein` (simple bean) from `Protein` (heavyweight class that implements many MS2 concepts)
- Move `ProteinManager` (except for one method) and `ProteinServiceImpl` (except for two methods) to Protein module
- Move `GoHelpers` back to MS2 module

### CODE REVIEW

File: ms2/src/org/labkey/ms2/protein/Protein.java [hunk 1]
```diff
@@ -44,16 +45,19 @@
...
 
-    private double _mass;
-    private String _sequence;
-    private int _seqId;
-    private String _description;
-    private String _bestName;
-    private String _bestGeneName;
+    public Protein(SimpleProtein simpleProtein)
+    {
+        setSeqId(simpleProtein.getSeqId());
```
Comment (labkey-jeckels) [inline ms2/src/org/labkey/ms2/protein/Protein.java]: Suggest moving this into a new SimpleProtein constructor and making this a simple pass-through.

File: ms2/src/org/labkey/ms2/protein/tools/GoHelpers.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            }
+        }
+    }
+
+    @Nullable
+    public static GoTypes GTypeStringToEnum(String label)
+    {
+        for (GoTypes g : GoTypes.values())
+        {
+            if (label.equals(g.toString())) return g;
```
Comment (labkey-jeckels) [inline ms2/src/org/labkey/ms2/protein/tools/GoHelpers.java]:
```suggestion
            if (g.toString().equals(label)) return g;
```

**Review by labkey-jeckels — APPROVED** [review]

File: ms2/src/org/labkey/ms2/protein/tools/GoHelpers.java (hunk 2 above)
Comment (labkey-adam) [reply ms2/src/org/labkey/ms2/protein/tools/GoHelpers.java]: Done

File: ms2/src/org/labkey/ms2/protein/Protein.java (hunk 1 above)
Comment (labkey-adam) [reply ms2/src/org/labkey/ms2/protein/Protein.java]: Done

**Pull request merged**